### PR TITLE
define struct types instead of using anonymous structs

### PR DESCRIPTION
**Pull request by fho**
Some fields were defined inline as anonymous structs.
This made it tedious to use these structs in testcases, because
the anonymous type needed to be redefined (incl the exact same JSON
tags) whenever it was assigned. 

Using anonymous struct also does not scale well when the struct changes.
For example when a field is added to an anonymous struct, every place
where the anonymous struct is assigned must also be changed.

Another advantage of using regular types is that accessors  will be
generated automatically and can be used instead of having to manually
check for nil values when accessing the struct fields.

Define the types as regular structs instead.

-----------

**PR Status:** PR is finished but I noticed that my company hasn't signed the CA yet. As soon as it is done, I'll transition the PR to "ready for review".

### CODE REVIEW

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:239]:
```suggestion
// EditRef represents a ref change of a pull-request.
```

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:244]:
```suggestion
// EditSHA represents a sha change of a pull-request.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
While you are here, do you want to do the same thing for:

* `ProjectChange`
* `ProjectCardChange`
* `ProjectColumnChange`
* `TeamChange`
* `RepositoryVulnerabilityAlertEvent`

?

**Comment by fho** [comment]
@gmlewis I also replaced the other anonymous struct types.

**Review by gmlewis — APPROVED** [review]
Thank you, @fho !
LGTM.

Awaiting second LGTM before merging.

**Comment by gmlewis** [comment]
Also, @fho - could you please remove the "Draft" setting when you are ready?

**Comment by fho** [comment]
@gmlewis I'll remove the draft status as soon as my employer signed the CA, probably beginning of next week. I only noticed that it was missing after I created it. :-)

**Comment by gmlewis** [comment]
> @gmlewis I'll remove the draft status as soon as my employer signed the CA, probably beginning of next week. I only noticed that it was missing after I created it. :-)

Interesting - the googlebot marked the PR as having passed the CLA test.  OK, sounds good.

**Comment by fho** [comment]
> > @gmlewis I'll remove the draft status as soon as my employer signed the CA, probably beginning of next week. I only noticed that it was missing after I created it. :-)
> 
> Interesting - the googlebot marked the PR as having passed the CLA test. OK, sounds good.

It's because I signed the PR for myself already in the past.

**Comment by fho** [comment]
@gmlewis my employer signed the CLA now also.
I can't see it on https://cla.developers.google.com/clas for my account but I'm in the contributors google-group of the CLA.

**Comment by gmlewis** [comment]
Friendly ping, @wesleimp

**Comment by sagar23sj** [comment]
If we are expecting to replace all anonymous structs with defined struct types, then I guess a few are still missing in this PR.
I am listing them below :
1. File: github.go / ErrorResponse
2. File: activity.go / Feeds

@gmlewis

**Comment by fho** [comment]
@sagar23sj thanks, I'll update the PR and also replace those anonymous structs.

It was not my initial intent of the PR to replace all anonymous structs but it makes sense to have a consistent solution

**Comment by fho** [comment]
done

**Review by gmlewis — APPROVED** [review]
Thank you, @fho !
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Comment by gmlewis** [comment]
@fho  - would you like to resolve the merge conflicts, or would you like me to do that?  I'm fine either way.

**Comment by fho** [comment]
@gmlewis also both fine with me, feel free to do changes :-)
Otherwise I can solve the conflict tomorrow

**Pull request merged**
